bindings/rust/s2n-tls/src/session_ticket.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::ptr::NonNull;
+use std::time::Duration;
+
+use s2n_tls_sys::*;
+
+use crate::{
+    connection::Connection,
+    error::{Error, Fallible},
+};
+
+/// A trait to retrieve session tickets from the connection
+pub trait SessionTicketCallback {
+    fn on_session_ticket(&self, connection: &mut Connection, session_ticket: SessionTicket);
+}
+
+/// Wrapper around a session ticket struct
+#[derive(Debug)]
+pub struct SessionTicket {
+    data: Vec<u8>,
+    lifetime: Option<Duration>,
+}
+
+impl SessionTicket {
+    pub fn from_raw(ticket: NonNull<s2n_session_ticket>) -> Result<Self, Error> {
+        let mut lifetime = 0;
+        unsafe { s2n_session_ticket_get_lifetime(ticket.as_ptr(), &mut lifetime).into_result()? };
+        let mut data_len = 0;
+        unsafe { s2n_session_ticket_get_data_len(ticket.as_ptr(), &mut data_len).into_result()? };
+        let mut data = vec![0; data_len];
+        unsafe {
+            s2n_session_ticket_get_data(ticket.as_ptr(), data_len, data.as_mut_ptr())
+                .into_result()?
+        };
+
+        Ok(Self {
+            lifetime: Some(Duration::new(lifetime.into(), 0)),
+            data,
+        })
+    }
+    pub fn session_lifetime(&self) -> Option<Duration> {
+        self.lifetime
+    }
+
+    pub fn session_data(&self) -> &Vec<u8> {
+        &self.data
+    }
+
+    pub fn new(data: Vec<u8>) -> Self {
```

Comment:
When would this function be used?

Reply:
If set_session takes a SessionTicket, but the user never stored the SessionTicket type and just stored the ticket bytes, then they have to construct a session ticket later when they call set_session(). 
See this test and its callback implementation in particular: https://github.com/aws/s2n-tls/pull/4114/files#diff-227eff6e31c2ecd83b6aabc74a0be84accb31a573ae28cab021bc6f65646bd83R217
This is the same reason why lifetime is an Option(see your above comment.) The user doesn't input the lifetime because s2n-tls doesn't actually need it.